File: backend/scripts/sqlite_export.php
<?php

if ($argc !== 2) {
    fwrite(STDERR, "Usage: sqlite_export.php <sqlite-db-path>\n");
    exit(1);
}

$dbPath = realpath($argv[1]);
if ($dbPath === false || !file_exists($dbPath)) {
    fwrite(STDERR, "SQLite database not found: {$argv[1]}\n");
    exit(1);
}

$db = new SQLite3($dbPath, SQLITE3_OPEN_READONLY);
$tablesResult = $db->query("
    SELECT name
    FROM sqlite_master
    WHERE type = 'table' AND name NOT LIKE 'sqlite_%'
    ORDER BY name
");

$payload = ['tables' => []];
while ($tableRow = $tablesResult->fetchArray(SQLITE3_ASSOC)) {
    $tableName = $tableRow['name'];
    $rowsResult = $db->query('SELECT * FROM "' . str_replace('"', '""', $tableName) . '"');
    $rows = [];
    while ($row = $rowsResult->fetchArray(SQLITE3_ASSOC)) {
        $rows[] = $row;
    }
    $payload['tables'][$tableName] = $rows;
}

echo json_encode($payload, JSON_UNESCAPED_UNICODE);

File: backend/scripts/sqlite_export.py
import json
import sqlite3
import sys
from pathlib import Path


def main() -> int:
    if len(sys.argv) != 2:
        print("Usage: sqlite_export.py <sqlite-db-path>", file=sys.stderr)
        return 1

    db_path = Path(sys.argv[1]).resolve()
    if not db_path.exists():
        print(f"SQLite database not found: {db_path}", file=sys.stderr)
        return 1

    conn = sqlite3.connect(f"file:{db_path}?mode=ro", uri=True)
    conn.row_factory = sqlite3.Row
    try:
        table_rows = conn.execute(
            """
            SELECT name
            FROM sqlite_master
            WHERE type = 'table' AND name NOT LIKE 'sqlite_%'
            ORDER BY name
            """
        ).fetchall()

        payload = {"tables": {}}
        for row in table_rows:
            table_name = row["name"]
            rows = conn.execute(f'SELECT * FROM "{table_name}"').fetchall()
            payload["tables"][table_name] = [dict(entry) for entry in rows]

        json.dump(payload, sys.stdout, ensure_ascii=False)
        return 0
    finally:
        conn.close()


if __name__ == "__main__":
    raise SystemExit(main())
